crates/rattler_conda_types/src/channel/mod.rs

```diff
@@ -444,11 +443,27 @@ pub(crate) const fn default_platforms() -> &'static [Platform] {
 
 /// Returns the specified path as an absolute path
 fn absolute_path(path: &str, root_dir: &Path) -> Result<Utf8TypedPathBuf, ParseChannelError> {
+    // Non parsable path
+    if path.starts_with("~\\") {
+        return Err(ParseChannelError::InvalidPath(path.to_owned()));
+    }
+
     let path = Utf8TypedPath::from(path);
     if path.is_absolute() {
         return Ok(path.normalize());
     }
 
+    // Parse the `~/` as the home folder
+    if let Ok(user_path) = path.strip_prefix("~/") {
+        return Ok(Utf8TypedPathBuf::from(
+            dirs::home_dir()
+                .ok_or(ParseChannelError::InvalidPath(path.to_string()))?
+                .to_string_lossy()
```

Comment:
You should use `to_str` and if that fails return `ParseChannelError::NotUtf8RootDir(..)`

Reply:
I changed it to return the given path. But the error is not really correct then, as it is going to error on the `dirs::home_dir()` not being available. Which is just weird and not well documented when that would go wrong.